scripts/base/protocols/dns/main.zeek

```diff
@@ -359,6 +367,13 @@ hook DNS::do_reply(c: connection, msg: dns_msg, ans: dns_answer, reply: string)
 		# Currently only standard queries are tracked.
 		return;
 
+	# Decode AD bit according to RFC 2535 if Z is set.
+	# Leave Z untouched in case existing users post-process the number value.
+	if ( msg$Z != 0 ) 
```

Comment:
IMO we should adapt the analyzer to decode the bits and then extend `dns_msg` with AD and CD fields (possibly always set to whatever Z would evaluate to rather than making them optional). This would give script writers immediate access to the fields without relying on logic in `dns/main.zeek`.

Extending dns.log with these new fields could be discussed separately (not sure the original ticket assumed that would be done). These fields might be useful enough in every deployment, but could also see not adding them (or not marking them `&log`) by default. But suggest to add them for now.

Reply:
Decoding the bits directly in the parser SGTM, and makes sense as these flags should be parsed out the same as RA/RD/etc - will make that change. Unless someone thinks otherwise, I'll still leave `Z` untouched, on the off chance someone downwstream depends on it holding all three bits. 

I'll still add them to the log - IMO, we log every other standard part of the DNS header, so we should log these as well.

Reply:
I agree with all here but this last bit is unusual — we don't normally leave a field in a message header and then parse out its bits in addition. I would normally vote for deprecation of the old `Z` field and introduction of a new one. On the other hand this case seems a bit special because even popular DNS resources often don't mention AD & CD.

 So ... in the end I agree — let's keep the Z field as-is, but let's also be very clear in the docstrings about how Z, AD, and CD relate in our implementation.

Reply:
Done - newly pushed changes moved the bit-parsing to the protocol parser. Also pushed a change to docs that explicitly calls out that Z is the 3-bit Z specified in the original DNS RFC.